providers/providers_manager.go

```diff
@@ -8,6 +8,8 @@ import (
 	"time"
 
 	"github.com/libp2p/go-libp2p-core/peer"
+	peerstore "github.com/libp2p/go-libp2p-core/peerstore"
+	peerstore_legacy "github.com/libp2p/go-libp2p-peerstore"
```

Comment:
nit: rename to something like peerstoreImpl or pstore. It's not quite a legacy repo since it has the peerstore implementations in it